test/OperationTest.cpp

```diff
@@ -79,3 +83,67 @@ TEST(OperationTest, getResultOnlyCached) {
   // tests.
   qec->getQueryTreeCache().clearAll();
 }
+
+// _____________________________________________________________________________
+
+MATCHER_P2(HasJsonKeyValue, key, value, "") {
+  try {
+    auto json = nlohmann::json::parse(arg);
+    if (!json.contains(key)) {
+      *result_listener << "JSON object does not contain key \"" << key << '"';
+      return false;
+    }
+    if (json[key] == value) {
+      return true;
+    }
+    *result_listener << "JSON key \"" << key << "\" has value \"" << json[key]
+                     << '"';
+  } catch (const nlohmann::json::parse_error& error) {
+    *result_listener << "failed to parse JSON";
+  }
+  return false;
+}
```

Comment:
This matcher is too generic and probably useful to let it go to waste in the `.cpp` file
Maybe add it to the `GTestHelpers.h` file (where we also have the generic `AD_EXPECT_THROW_WITH_MESSAGE and add a short comment and example.